### PR TITLE
Rest api thread can raise an exception during shutdown

### PR DESCRIPTION
**Pull request by CyberDem0n**
catch it and report

### CODE REVIEW

**Comment by alexeyklyukin** [comment]
👍

File: patroni/__init__.py [hunk 1]
```diff
@@ -129,7 +129,10 @@ def setup_signal_handlers(self):
         signal.signal(signal.SIGTERM, self.sigterm_handler)
 
     def shutdown(self):
-        self.api.shutdown()
+        try:
+            self.api.shutdown()
+        except Exception:
+            logger.exception('Exception durig RestApi.shutdown')
```
Comment (alexeyklyukin) [inline patroni/__init__.py]: during.

Comment (alexeyklyukin) [reply patroni/__init__.py]: also want to see the specifics on the exception

Comment (CyberDem0n) [reply patroni/__init__.py]: what specifics?
logger.exception will show the full stacktrace

**Comment by alexeyklyukin** [comment]
👍

**Comment by avaczi** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**